Feature: <Toolbar> component static version

ye-jin-jang opened the pull request:
### [stsatic]<Toolbar /> 스태틱 버전 구현

### Task

-[task card 링크](https://www.notion.so/vanillacoding/7ddc16eec4394b469b80ceae0d31c7e5?v=b86410c7460448958feaf40b176cd9e4&p=f295abf15b82406bba791f3ebbf59fe1&pm=s)

### Description
```
 <Toolbar>
 <PenStatusTool>
 <EraserStatusTool>
 <HightLightStatusTool>
 <PostIt>
```
1. Toolbar 컴포넌트에 mode 변수 값에 따라 해당 PenStatusTool 컴포넌트, EraserStatusTool 컴포넌트, HightLightStatusTool 컴포넌트, PostIt 컴포넌트로 변경되도록 mode 상태변수 하나만 추가했습니다.

2. 마우스를 아이콘에 다가갔을 경우 그림자 스타일 적용

3. 아이콘 선택버전과 미선택버전 아이콘png들은 /assets/icon 파일에 있습니다.

### Point

- 컴포넌트나 styled component 이름이 적절한지 확인

### ETC

아이콘을 클릭했을 경우 아이콘 변경 및 클릭음영은 추가하지 않았습니다.

### PR 전 확인사항

- [X] 가장 최신 브랜치를 pull했습니다.
- [X] base 브랜치명을 확인했습니다.(Front, Backend, feature ...)
- [X] 코드 컨벤션을 모두 지켰습니다.
- [X] 적절한 라벨이 있습니다.
- [X] assignee가 있습니다.

### 스크린샷
PenStatusTool 컴포넌트
![image](https://user-images.githubusercontent.com/78009532/225611173-4d5262e3-d94f-4546-b5ac-473feb7e7327.png)

EraserStatusTool 컴포넌트
![image](https://user-images.githubusercontent.com/78009532/225610566-ee963ba0-5238-458a-b03e-f92e6dad99ca.png)

HightLightStatusTool 컴포넌트
![image](https://user-images.githubusercontent.com/78009532/225610243-36323714-d3c1-4606-8abf-636bdcdcc89c.png)

PostIt 컴포넌트
![image](https://user-images.githubusercontent.com/78009532/225608695-3af5e826-0dee-487e-a4ef-132ffb99ef3c.png)

ye-jin-jang committed:
Feat. <Toolbar> 컴포넌트 정적버전 구현
pen컴포넌트, eraser컴포넌트, hightlight컴포넌트, poset-it컴포넌트 정적 버전 구현

JisookDev commented on package.json:
```diff
@@ -3,12 +3,16 @@
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^6.3.0",
+    "@fortawesome/free-solid-svg-icons": "^6.3.0",
+    "@fortawesome/react-fontawesome": "^0.2.0",
```
fortawesome을 어떤 목적으로 사용하셨는지 궁금합니다.👀
저는 SVG 또는 img를 syled component에서 image 태그로 불러와서 작업해도 문제가 없어서.. 
다같이 논의해보면 좋겠습니다 ~

JisookDev commented on src/components/EraserStatusTool.js:
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import styled from "styled-components";
```
라이브러리와 내부 불러오는 파일 사이에 개행이 필요합니다!

JisookDev commented on src/components/EraserStatusTool.js:
```diff
@@ -0,0 +1,32 @@
+const ToolStatusField = styled.div`
+  display: flex;
+  width: 50%;
+  border: 1px solid black;
```
컨벤션 규칙에 따라 위치 관련 속성(display / align-items / justify-content)은 다같이 최상단에 위치하는게 좋을 것 같아요!

JisookDev commented on src/components/HightLightStatusTool.js:
```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import styled from "styled-components";
```
개행 👀

JisookDev commented on src/components/HightLightStatusTool.js:
```diff
@@ -0,0 +1,53 @@
+const ToolStatusField = styled.div`
+  display: flex;
+  width: 50%;
+  border: 1px solid black;
```
위치 컨벤션 속성 최상단 👀

JisookDev commented on src/components/HightLightStatusTool.js:
```diff
@@ -0,0 +1,53 @@
+  justify-content: space-around;
+`;
+
+const Color = styled.div`
```
`<Color type="color" value="#FFFF00" />`로 하고 const Color = styled.input 으로 해도 되지 않을까요??

JisookDev commented on src/components/PenStatusTool.js:
```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import styled from "styled-components";
```
개행 👀

JisookDev commented on src/components/Toolbar.js:
```diff
@@ -0,0 +1,67 @@
+import PostIt from "./PostIt";
+
+export default function Toolbar() {
+  const [mode, setMode] = useState("postit");
```
postit도 camel case로 작성해야할 것 같아요..!

JisookDev commented on src/components/Toolbar.js:
```diff
@@ -0,0 +1,67 @@
+import React, { useState } from "react";
+import styled from "styled-components";
```
개행 👀

hweejae-shin commented on src/components/Toolbar.js:
```diff
@@ -0,0 +1,67 @@
+import pen1Icon from "../assets/icon/pen1.png";
+import eraser2Icon from "../assets/icon/eraser2.png";
+import marker2Icon from "../assets/icon/marker2.png";
+import post2Icon from "../assets/icon/post2.png";
```
아이콘 만드시느라 고생하셨습니다!
../assets/icon 폴더에서 여러 png파일을 불러오는데,
import { pen2Icon, eraser1Icon } 처럼 구조분해 할당으로 가져올 수 있다면, 이 방식이 가독성이 더 좋지 않을까 싶습니다!

JisookDev replied:
가능한지는 잘 모르겠지만, 해당 아이콘들이 모두 /assets/icon/이라는 폴더에서 불러오는데 '구조분해할당'으로 한줄로 수정하는건 어떨까요..?🤔

hweejae-shin commented on src/components/Toolbar.js:
```diff
@@ -0,0 +1,67 @@
+
+export default function Toolbar() {
+  const [mode, setMode] = useState("postit");
+  let editorTool = null;
```
작업도구가 바뀜에 따라 해당 상태 툴이 올바르게 마운트 되는지 점검하시느라 쓰신 코드 같습니다!
실제로 상태가 적용되면, zustand를 사용해 전역으로 관리될테니 향후 작업자는 useState를 꼭 안써도 될 것 같네요.

hweejae-shin reviewed (commented):
툴바를 동적으로 만들어야해서 고려하실 사항이 많으셨을텐데, 고생 많으셨습니다!

나중에 상태 반영할때 아주 수월해질 것 같습니다!

ye-jin-jang committed:
Design. 코드 리뷰 반영
png 파일들을 export하는 editorIcon.js를 만들어 컴포넌트에서 가독성있게 import하도록 구현, 코드 컨벤션에 맞게 코드 수정

ye-jin-jang committed:
Fix. App.js 필요없는 코드 제거
컴포넌트 테스트를 위해 작성했던 컴포넌트 제거

ye-jin-jang committed:
Style. camel case로 변경
Toolbar 컴포넌트에서 postit -> postIt 으로 변경

ye-jin-jang committed:
Merge branch 'development' of https://github.com/Thinking-in-human/Noted-client into feature/toolbar

JisookDev commented on src/components/PenStatusTool.js:
```diff
@@ -0,0 +1,58 @@
+        <WidthIcon type="image" src={width2Icon} alt="widthBasicIcon" />
+        <WidthIcon type="image" src={width3Icon} alt="widthThickIcon" />
+      </FormatIcon>
+      <input type="range" min="2" max="40" value="2" />
```
css가 적용되지 않는 태그는 HTML 태그로 작성하겠습니다~

ye-jin-jang committed:
Style. postIt 컴포넌트 rename
postIt -> postItStatusTool로 변경한 후 Toolbar 컴포넌트 수정
